spring-integration-file/src/main/java/org/springframework/integration/file/config/AbstractRemoteFileInboundChannelAdapterParser.java

```diff
@@ -94,4 +99,6 @@ protected final BeanMetadataElement parseSource(Element element, ParserContext p
 
 	protected abstract Class<? extends FileListFilter<?>> getRegexPatternFileListFilterClass();
 
+	protected abstract Class<? extends AbstractPersistentAcceptOnceFileListFilter<?>> getPersistentAcceptOnceFileListFilterClass();
```

Comment:
Perhaps return null by default? (Allow subclasses, e.g. AWS to not provide a filter).

Maybe not - since this is a major release, I guess it's ok.